File: mobile/lib/models/shop.dart
class Shop {
  final String id;
  final String name;
  final String occupation;
  final int totalSeats;
  final int avgTimePerCustomer;
  final bool isAcceptingOnline;
  final String address;
  final String description;
  final String? qrCodeString;

  const Shop({
    required this.id,
    required this.name,
    required this.occupation,
    required this.totalSeats,
    required this.avgTimePerCustomer,
    required this.isAcceptingOnline,
    this.address = '',
    this.description = '',
    this.qrCodeString,
  });

  factory Shop.fromJson(Map<String, dynamic> json) {
    return Shop(
      id: (json['_id'] ?? json['id']) as String,
      name: json['name'] as String,
      occupation: json['occupation'] as String? ?? 'General Service',
      totalSeats: json['totalSeats'] as int,
      avgTimePerCustomer: json['avgTimePerCustomer'] as int,
      isAcceptingOnline: json['isAcceptingOnline'] as bool? ?? true,
      address: json['address'] as String? ?? '',
      description: json['description'] as String? ?? '',
      qrCodeString: json['qrCodeString'] as String?,
    );
  }
}

File: backend/src/models/Shop.js
const mongoose = require('mongoose');

const shopSchema = new mongoose.Schema({
  ownerId: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
  name: { type: String, required: true, trim: true },
  occupation: { type: String, required: true, trim: true },
  description: { type: String, default: '' },
  address: { type: String, default: '' },
  totalSeats: { type: Number, required: true, min: 1 },
  avgTimePerCustomer: { type: Number, required: true, min: 1 },
  isAcceptingOnline: { type: Boolean, default: true },
  qrCodeString: { type: String },
  openTime: { type: String, default: '09:00' },
  closeTime: { type: String, default: '18:00' },
  createdAt: { type: Date, default: Date.now },
});

shopSchema.pre('save', function (next) {
  if (!this.qrCodeString) {
    this.qrCodeString = `noqeu://shop/${this._id}`;
  }
  next();
});

module.exports = mongoose.model('Shop', shopSchema);
